### PR TITLE
Update test_miri_lrs_slitless to chain tso1 to tso2

### PR DESCRIPTION
**Pull request by jdavies-st**
This chains together the tso1 pipeline with the tso2 and tso3 pipelines using the same dataset from the tso1 fixture.

All the truth files have been updated.

### CODE REVIEW

File: jwst/regtest/test_miri_lrs_slitless.py [hunk 1]
```diff
@@ -57,20 +54,18 @@ def generate_tso3_asn():
     """Generate an association file that references the output of run_spec2_pipeline."""
     asn = asn_from_list([f"{DATASET_ID}_calints.fits"], product_name=PRODUCT_NAME)
     asn.data["program"] = "80600"
-    asn.data["asn_id"] = "o012"
     asn.data["asn_type"] = "tso-spec3"
```
Comment (hbushouse) [inline jwst/regtest/test_miri_lrs_slitless.py]: This isn't a valid ASN type. It should be just "tso3".

File: jwst/regtest/test_miri_lrs_slitless.py [hunk 2]
```diff
@@ -83,17 +78,14 @@ def run_spec3_pipeline(run_spec2_pipeline, generate_tso3_asn):
...
 
 @pytest.mark.bigdata
-@pytest.mark.parametrize("step_suffix", ['rate', 'rateints', 'linearity',
-                                         'rscd',
-                                         'dq_init',
-                                         'saturation', 'dark_current', 'refpix',
-                                         'jump'])
-def test_miri_lrs_slitless_tso1(run_tso1_pipeline, fitsdiff_default_kwargs, step_suffix):
+@pytest.mark.parametrize("step_suffix", ['dq_init', 'saturation', 'refpix',
+    'rscd', 'linearity', 'dark_current', 'jump', 'rate', 'rateints',])
```
Comment (hbushouse) [inline jwst/regtest/test_miri_lrs_slitless.py]: Is the trailing comma OK? " 'rateints',])"

File: jwst/regtest/test_miri_lrs_slitless.py [hunk 3]
```diff
@@ -104,14 +96,14 @@ def test_miri_lrs_slitless_tso1(run_tso1_pipeline, fitsdiff_default_kwargs, step
 
 @pytest.mark.bigdata
 @pytest.mark.parametrize("step_suffix", ["flat_field", "srctype", "calints", "x1dints"])
-def test_miri_lrs_slitless_spec2(run_spec2_pipeline, rtdata_module, fitsdiff_default_kwargs,
+def test_miri_lrs_slitless_tso_spec2(run_tso_spec2_pipeline, rtdata_module, fitsdiff_default_kwargs,
     step_suffix):
     """Compare the output of a MIRI LRS slitless calwebb_tso-spec2 step."""
```
Comment (hbushouse) [inline jwst/regtest/test_miri_lrs_slitless.py]: Minor nit, but I'd change "calwebb_tso-spec2 **step**" to "calwebb_tso-spec2" **pipeline**.

File: jwst/regtest/test_miri_lrs_slitless.py [hunk 4]
```diff
@@ -57,20 +54,18 @@ def generate_tso3_asn():
     """Generate an association file that references the output of run_spec2_pipeline."""
     asn = asn_from_list([f"{DATASET_ID}_calints.fits"], product_name=PRODUCT_NAME)
     asn.data["program"] = "80600"
```
Comment (eslavich) [inline jwst/regtest/test_miri_lrs_slitless.py]: I realize this isn't part of this PR, but I'd recommend putting this at the top as a constant -- down here, it's easy for it to fall out of sync with the chosen dataset (as it seems to have done already).

File: jwst/regtest/test_NIRcam_level2a.py [hunk 5]
```diff
@@ -1,100 +0,0 @@
-""" Test for the detector1 pipeline using NIRCam data. This should take an uncal
```
Comment (hbushouse) [inline jwst/regtest/test_NIRcam_level2a.py]: So where did this test file go? I see that the NIRSpec_IRS2 test module was renamed and included below, but I don't see any replacement for this NIRCam test module.

Comment (jdavies-st) [reply jwst/regtest/test_NIRcam_level2a.py]: I deleted it.  We already have a detector1 pipeline test for NIRCam imaging data with better simulated input data in the chained pipelines test, `test_nircam_image.py`.

**Pull request merged**